core/templates/pages/topic-editor-page/modal-templates/create-new-subtopic-modal.component.spec.ts

```diff
@@ -0,0 +1,254 @@
+    type: 'html',
+    ui_config: {
+      rows: 100
+    }
+  };
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        CreateNewSubtopicModalComponent
+      ],
+      providers: [
+        {
+          provide: WindowRef,
+          useClass: MockWindowRef
+        },
+        {
+          provide: NgbActiveModal,
+          useClass: MockActiveModal
+        },
+        {
+          provide: TopicEditorStateService,
+          useClass: MockTopicEditorStateService
+        },
+        TopicUpdateService,
+        SubtopicValidationService
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateNewSubtopicModalComponent);
+    component = fixture.componentInstance;
+
+    ngbActiveModal = TestBed.inject(NgbActiveModal);
+    TestBed.inject(WindowRef);
+    topicUpdateService = TestBed.inject(TopicUpdateService);
+    topicEditorStateService =
+      (TestBed.inject(TopicEditorStateService) as unknown) as
+      jasmine.SpyObj<MockTopicEditorStateService>;
+    subtopicValidationService = TestBed.inject(SubtopicValidationService);
+    topicObjectFactory = TestBed.inject(TopicObjectFactory);
+
+    topic = topicObjectFactory.createInterstitialTopic();
+    let subtopic1 = Subtopic.createFromTitle(1, 'Subtopic1');
+    topic.getSubtopics = function() {
+      return subtopic1;
+    };
+    topic.getId = function() {
+      return '1';
+    };
+    topic.getNextSubtopicId = function() {
+      return 1;
+    };
+    spyOn(topicEditorStateService, 'getTopic').and.returnValue(topic);
+
+    fixture.detectChanges();
+  });
+
+  it('should assign default values to modal', () => {
+    spyOn(topicUpdateService, 'addSubtopic').and.stub();
+
+    component.ngOnInit();
+    expect(component.SUBTOPIC_PAGE_SCHEMA).toEqual(DefaultSubtopicPageSchema);
+    expect(component.subtopicId).toBe(1);
+    expect(component.hostname).toBe('local');
+    expect(component.classroomUrlFragment).toBe('non');
+    expect(component.schemaEditorIsShown).toBe(false);
+    expect(component.subtopicUrlFragmentExists).toBe(false);
+    expect(component.errorMsg).toBe(null);
+    expect(component.MAX_CHARS_IN_SUBTOPIC_TITLE)
+      .toBe(constants.MAX_CHARS_IN_SUBTOPIC_TITLE);
+    expect(component.MAX_CHARS_IN_SUBTOPIC_URL_FRAGMENT)
+      .toBe(constants.MAX_CHARS_IN_SUBTOPIC_URL_FRAGMENT);
+    expect(topicUpdateService.addSubtopic).toHaveBeenCalledWith(topic, '');
+  });
+
+  it('should call getSchema function properly', () => {
+    let SUBTOPIC_PAGE_SCHEMA = component.getSchema();
+
+    expect(SUBTOPIC_PAGE_SCHEMA).toEqual(DefaultSubtopicPageSchema);
+  });
+
+  it('should call showSchemaEditor function properly', () => {
+    component.showSchemaEditor();
+
+    expect(component.schemaEditorIsShown).toBe(true);
+  });
+
+  it('should call updateSubtopicThumbnailFilename function properly', () => {
```

Comment:
again, the test name does not suggest what it is actually doing. Test message should tell exactly what is being done in the test. You may follow the 'should do X if Y/ when Y happens' method here as well.

Please refer to the FE test guide in the wiki too [here](https://github.com/oppia/oppia/wiki/Frontend-tests#write-frontend-tests)

Reply:
DONE